### PR TITLE
EVG-15100 Fallback to cedar test results for test count resolvers

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-15100](https://jira.mongodb.org/browse/EVG-15100)

### CODE REVIEW

**Review by arjunpatel100 — APPROVED** [review]

File: graphql/resolvers.go [hunk 1]
```diff
@@ -2397,18 +2397,67 @@ func (r *taskResolver) EstimatedStart(ctx context.Context, obj *restModel.APITas
 	return &duration, nil
 }
 func (r *taskResolver) TotalTestCount(ctx context.Context, obj *restModel.APITask) (int, error) {
-	tests, err := r.sc.GetTestCountByTaskIdAndFilters(*obj.Id, "", nil, obj.Execution)
+	testCount, err := r.sc.GetTestCountByTaskIdAndFilters(*obj.Id, "", nil, obj.Execution)
 	if err != nil {
 		return 0, InternalServerError.Send(ctx, fmt.Sprintf("Error getting test count: %s", err.Error()))
 	}
-	return tests, nil
+	// Check cedar if we dont have any test results in the db
```
[the remark this answers is not in the document]
Comment (ablack12) [reply graphql/resolvers.go:2404]: This wording is actually consistent with Evergreen's usage.

File: graphql/resolvers.go [hunk 2]
```diff
@@ -2397,18 +2397,67 @@ func (r *taskResolver) EstimatedStart(ctx context.Context, obj *restModel.APITas
...
+		if len(obj.ExecutionTasks) > 0 {
+			opts.DisplayTaskID = *obj.Id
+		} else {
+			opts.TaskID = *obj.Id
+		}
+
+		cedarTestResults, err := apimodels.GetCedarTestResults(ctx, opts)
+		if err != nil {
+			grip.Warning(message.WrapError(err, message.Fields{
+				"task_id": obj.Id,
```
Comment (ablack12) [inline graphql/resolvers.go:2420]: Should we log *obj.Id or will this still print correctly?

**Review by ablack12 — COMMENTED** [review]
It seems like in other places we do this the opposite way (fallback to DB only if cedar test results don't exist). Is there a reason we're doing it the opposite way here? Since most projects have moved to cedar it does seem like the right order.

**Comment by khelif96** [comment]
> It seems like in other places we do this the opposite way (fallback to DB only if cedar test results don't exist). Is there a reason we're doing it the opposite way here? Since most projects have moved to cedar it does seem like the right order.

No particular reason. I just swapped the order to make it more consistent with the rest of the code base.

**Review by ablack12 — APPROVED** [review]

**Pull request merged**